onnxruntime/core/providers/webgpu/reduction/reduction_ops.h

```diff
@@ -39,23 +40,41 @@ class ReduceKernel : public WebGpuKernel, public ReduceKernelBase<allow_multi_ax
   using ReduceKernelBase<allow_multi_axes>::keepdims_;
   using ReduceKernelBase<allow_multi_axes>::select_last_index_;
 
-  ReduceKernel(const OpKernelInfo& info, std::string name, optional<int64_t> keepdims_override = {})
+  ReduceKernel(const OpKernelInfo& info, std::string name, bool allow_empty_input = false, optional<int64_t> keepdims_override = {})
       : WebGpuKernel(info),
         ReduceKernelBase<allow_multi_axes>(info, keepdims_override),
-        name_(name) {
+        name_(name),
+        allow_empty_input_(allow_empty_input) {
   }
   Status ComputeInternal(ComputeContext& ctx) const;
-  virtual ReduceOpSpecificCode GetOpSpecificCode(const Tensor* input_tensor, size_t axes_size) const = 0;
+  virtual ReduceOpSpecificCode GetOpSpecificCode(const Tensor* input_tensor) const = 0;
+
+  Status CheckInput(const Tensor* input_tensor) const {
```

Comment:
I think the check input functions should be no longer needed in ORT EP. ORT should have already checked the graph against kernel defs.

Reply:
In this case we are checking something that is not already checked based on the schema. For sum Reduce operations, such as ReduceSum, ReduceSquareSum, ReduceL1, ReduceL2, etc. it is legal to use a tensor with zero size along the reduce dims. However For ReduceMax and ReduceMin it is not legal. Strangely one test case for ReduceSum accepts even when the output has a zero, dims.